openmdao/components/exec_comp.py

```diff
@@ -339,6 +341,16 @@ def _setup_expressions(self):
                                        "recognized for variable '%s': %s" %
                                        (self.msginfo, arg, sorted(diff)))
 
+                if 'val' in val and 'value' in val:
+                    raise RuntimeError("Cannot use 'val' and 'value' at the same time, use 'val'.")
```

Comment:
I think you need `self.msginfo` in this warning message to point the user to the component in question.